E2E test improvements and consolidatng

alexflorisca opened the pull request:
There's been a few PRs on E2E tests and as such there's been a bit of cross over. This PR aims to simplify things a bit. Specifically:

- Use `useTheme()` helper to make sure we are remembering and switching back to the original theme after activating a new theme within a test
- Move all block related functions under the `shopper.block` object for clarity
- Simplify the `goToBlockPage()` function to derive the block page permalink directly, rather than visiting wp-admin
- Remove now unnecessary constants

### Testing

#### Automated Tests
* [x] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

### Manual Testing

How to test the changes in this Pull Request:

1. Make sure all E2E tests pass

<!-- If you can, add the appropriate labels -->

Commit:
Refactor e2e tests slightly for better efficiency

tarhi-saad commented on tests/utils/shopper.js:
```diff
@@ -7,71 +7,60 @@ import {
+		// the page title so we can derive them directly
+		goToBlockPage: async ( blockName ) => {
+			const config = require( 'config' );
+			const baseUrl = config.get( 'url' );
```
We can import the `BASE_URL` from `tests/e2e/utils.js` and replace:

```js
const config = require( 'config' );
const baseUrl = config.get( 'url' );
...
const url = baseUrl + pageTitle.toLowerCase().replace( / /g, '-' );
```

with

```js
/**
 * Internal dependencies
 */
import { BASE_URL } from '../utils';
...
const url = BASE_URL + pageTitle.toLowerCase().replace( / /g, '-' );
```

Commit:
Add "proceedToCheckout" shopper method & refactor code

Commit:
Use "beforeEach" in case we add more test cases

I've kept the "afterAll" function to reset any side effects at the end
of this file's tests

Commit:
Use "beforeEach" in "checkout-place-order-as-guest" file

tarhi-saad reviewed (approved):
Thank you, @alexflorisca, for this great PR! Everything looks good. There is just a tiny suggestion I made below. And I also pushed some commits where I implemented some suggestions you made in my latest PR.

I've used `beforeEach` as you suggested in `cart-proceed-to-checkout.test.js`, but kept `afterAll` to reset the last test in this file to avoid any left side effects at the end. Please let me know your thoughts on this.

Commit:
Use BASE_URL in shopper.block.goToBlockPage

Commit:
Merge branch 'update/test-refactor' of github.com:woocommerce/woocommerce-gutenberg-products-block into update/test-refactor
